core/src/gl/dynamicQuadMesh.h

```diff
@@ -70,35 +94,64 @@ void DynamicQuadMesh<T>::upload(RenderState& rs) {
 }
 
 template<class T>
-bool DynamicQuadMesh<T>::draw(RenderState& rs, ShaderProgram& _shader) {
+bool DynamicQuadMesh<T>::draw(RenderState& rs, ShaderProgram& shader) {
+    return draw(rs, shader, 0);
+}
 
+template<class T>
+bool DynamicQuadMesh<T>::draw(RenderState& rs, ShaderProgram& shader, int textureUnit) {
     if (m_nVertices == 0) { return false; }
 
     // Bind buffers for drawing
     rs.vertexBuffer(m_glVertexBuffer);
     rs.indexBuffer(rs.getQuadIndexBuffer());
 
     // Enable shader program
-    if (!_shader.use(rs)) {
+    if (!shader.use(rs)) {
         return false;
     }
 
-    size_t vertexOffset = 0;
-    size_t maxVertices = RenderState::MAX_QUAD_VERTICES;
-
-    for (size_t offset = 0; offset < m_nVertices; offset += maxVertices) {
-        size_t nVertices = maxVertices;
-
-        if (offset + maxVertices > m_nVertices) {
-            nVertices = m_nVertices - offset;
+    size_t verticesDrawn = 0;
```

Comment:
Great job with updates on variable names here. Makes the code much more clearer.